crates/solidity/inputs/language/bindings/rules.msgb

```diff
@@ -495,35 +527,35 @@ attribute symbol_reference = symbol  => type = "push_symbol", symbol = symbol, i
 
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;;; Function and parameter declarations
+;;; Function, parameter declarations and modifiers
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-@param [Parameter] {
+@param [Parameter @type_name [TypeName]] {
   node @param.lexical_scope
   node @param.def
-}
 
-@param [Parameter @type_name [TypeName]] {
   edge @type_name.type_ref -> @param.lexical_scope
-}
-
-@param [Parameter @type_name [TypeName] @name [Identifier]] {
-  node def
-  attr (def) node_definition = @name
-  attr (def) definiens_node = @param
 
-  edge @param.def -> def
+  node @param.typeof
+  attr (@param.typeof) push_symbol = "@typeof"
+  edge @param.typeof -> @type_name.output
+}
 
-  node typeof
-  attr (typeof) push_symbol = "@typeof"
+@param [Parameter @name [Identifier]] {
+  attr (@param.def) node_definition = @name
+  attr (@param.def) definiens_node = @param
 
-  edge def -> typeof
-  edge typeof -> @type_name.output
+  edge @param.def -> @param.typeof
 }
 
 @params [ParametersDeclaration] {
   node @params.lexical_scope
   node @params.defs
+
+  ;; This scope can be used to resolve named argument calls
+  node @params.args
```

Comment:
I'm a bit confused by the `args` node here, since "arguments" are on the caller expression side. Isn't this node on the parameter definition side?

Reply:
Hmm... you're right it's confusing. I'll rename this to `param_names` to better reflect the meaning of the defined scope.

Reply:
I settled for `@params.names` for function parameters. For errors/events I'm using `@event.params` or `@error.params` and the synthetic symbol into the scope is `@param_names` in both cases. On the arguments side, the scoped node is still called `@args.refs`.